vendor/bat-native-ads/include/bat/ads/ads.h

```diff
@@ -111,6 +117,51 @@ class ADS_EXPORT Ads {
       const NotificationInfo& info,
       const NotificationResultInfoResultType type) = 0;
 
+  // Should be called to retrieve ads history
+  virtual std::map<std::string, std::vector<AdsHistory>> GetAdsHistory() = 0;
+
+  // Should be called to indicate interest in the given ad. This is a
+  // toggle, so calling it again returns the setting to the neutral
+  // state
+  virtual AdContent::LikeAction ToggleAdThumbUp(

```

Comment:
@tmancey Just to fill you in on how these work, the action parameter tells what the action was pre-click. The return value is what the action is after toggling. Note that a thumbs up can toggle between thumbs up and neutral and a thumbs down can toggle between thumbs down and neutral. In other words, there are three total states, controlled by two different icons.